spring-cloud-stream-binder-kafka-streams/src/main/java/org/springframework/cloud/stream/binder/kafka/streams/KafkaStreamsStreamListenerSetupMethodOrchestrator.java

```diff
@@ -268,10 +271,21 @@ else if (arguments.length == 1 && StringUtils.hasText(inboundName)) {
 							.getExtendedConsumerProperties(inboundName);
 					// get state store spec
 					KafkaStreamsStateStoreProperties spec = buildStateStoreSpec(method);
-					Serde<?> keySerde = this.keyValueSerdeResolver
-							.getInboundKeySerde(extendedConsumerProperties);
-					Serde<?> valueSerde = this.keyValueSerdeResolver.getInboundValueSerde(
-							bindingProperties.getConsumer(), extendedConsumerProperties);
+
+					Serde<?> keySerde;
+					Serde<?> valueSerde;
+
+					if (bindingServiceProperties.getConsumerProperties(inboundName).isUseNativeDecoding()) {
+
+						keySerde = this.keyValueSerdeResolver
+								.getInboundKeySerde(extendedConsumerProperties, ResolvableType.forMethodParameter(methodParameter));
+						valueSerde = this.keyValueSerdeResolver.getInboundValueSerde(
+								bindingProperties.getConsumer(), extendedConsumerProperties, ResolvableType.forMethodParameter(methodParameter));
+					}
+					else {
+						keySerde = Serdes.ByteArray();
+						valueSerde = Serdes.ByteArray();
```

Comment:
Same serde comment as well as the fact that the code looks similar to the code above which may be better to be moved into something shared (e.g., utility class, abstract class etc)

Reply:
yep, this is going to be cleaned up as part of a refactoring PR after this.